### PR TITLE
round scaled dimensions up to avoid zero size

### PR DESCRIPTION
**Pull request by mexon**
This bug surfaced in my own plugin that scrapes content from websites, when it tries to insert an extremely non-square image such as https://ichef.bbci.co.uk/news/624/cpsprodpb/179B9/production/_106279669_fb1c6c40-2f65-4fc3-9432-c4967c0d59c7.jpg which is 624x2 pixels big.  It attempts to scale the image to a maximum of 90 pixels (hard-coded).  Using `intval`, this produces a height of 0 pixels.  Imagemagick then throws an exception trying to resize it.

This change rounds upwards.  It also includes an early return in the `scale()` function when asked to scale an image to zero size, treating this the same as an invalid image.

### CODE REVIEW

File: src/Util/Images.php
Comment (annando) [inline src/Util/Images.php:421]: Why `ceil`? This would always increase the height by 1, wouldn't it? Wouldn't it be much easier and safer to use für example `max(intval($height * $max) / $width), 1);`?

Comment (mexon) [reply src/Util/Images.php:421]: I guess it will mostly be one extra pixel compared to the existing approach, yeah.  But I don't see that as any less correct than what it's doing now, which is effectively `floor()`.  That is, `intval(479.9999)` will give `479`.

In general, as a user, to the extent I care about this at all, having one extra pixel gives me more image to look at.  `ceil()` very slightly biases the calculation to be very slightly "squarer", which is probably the right bias.

I do see adding an explicit rounding decision as a good thing for readability, rather than adding a special case for zero pixels.

And, if the input image really is 320x0, then the `max(..., 1)` approach will give a final height of 1.  Who knows what mess that might cause.

Comment (annando) [reply src/Util/Images.php:421]: I don't think that pictures can have `0` pixels, so this isn't an issue here. But you mentioned a good point. I guess that we should use the `round` function with this special case for the `0`. From a perfectionists standpoint I wouldn't appreciate that for example a `128x128` picture is resized to a `65x64` picture.

Comment (mexon) [reply src/Util/Images.php:421]: I still think that adding another special case here makes the code more complicated and harder to maintain.  Especially in the absence of unit tests, I'd worry someone would undo the fix later.  There is already a pretty mind-bending special case of > 16:9 aspect ratio images.  Using `round` instead of `ceil` with `$mode = PHP_ROUND_HALF_UP` seems reasonable though - I'm probably too much of a C-programmer at heart 😜

Comment (annando) [reply src/Util/Images.php:421]: We do have got unit tests and it seems as if it should be no problem to add a test for `getScalingDimensions` in https://github.com/friendica/friendica/blob/develop/tests/src/Util/ImagesTest.php

Comment (MatthewExon-TomTom) [reply src/Util/Images.php:421]: Yeah, you're right, I should do that before this gets merged. It especially could use some tests around the 16:9 special case too.

Comment (MrPetovan) [reply src/Util/Images.php:421]: Did you reply with your professional GitHub account? 👀

(Happened to me a few times)

Comment (annando) [reply src/Util/Images.php:421]: @MrPetovan off topic: Since I don't have got a professional GitHub account, I use this one for professional work as well. I think that a "my little pony" avatar is suitable for reputable work. :grin:

Comment (MrPetovan) [reply src/Util/Images.php:421]: I actually have two professional accounts, but I lost access to the first one because I lost access to my 2FA seed and had no other way of proving account ownership 🤷‍♂️

Comment (mexon) [reply src/Util/Images.php:421]: Argh yes - twice in one PR!  Thanks for the notice.  Not especially sensitive of course, but I do try to keep my employer's name out of it.  Anyway, the status is: I just got back from my travels, and on the weekend I plan to add unit tests.  This PR is on hold until that's done.

Comment (mexon) [reply src/Util/Images.php:421]: @annando could you give me a hint how to run unit tests locally?  My notes say the last time I tried this was 3 years ago, but that's not working any more:
```
mat@aeon:~/Project/friendica$ USER=friendica_test PASS=friendica_test DB=friendica_test sudo phpunit --configuration tests/phpunit.xml
PHP Fatal error:  Uncaught Error: Class "Friendica\Test\FixtureTest" not found in /home/mat/Project/friendica/tests/functional/DependencyCheckTest.php:36
```
The file `tests/README.md` points to a non-existent script I'm supposed to run.

Comment (MrPetovan) [reply src/Util/Images.php:421]: It looks like it's now called `bin/dev/autotest.sh`.

Comment (mexon) [reply src/Util/Images.php:421]: OK, thanks.  Tests added.

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
I approved the tests but they are currently failing, I expect this to be because tests were written before code was written to satisfy those tests.

We... aren't used to this way of interacting with tests here. 😅

**Comment by mexon** [comment]
> I approved the tests but they are currently failing, I expect this to be because tests were written before code was written to satisfy those tests.

Uh, no?  I mean, I'm a fan of test-driven development, but this ain't it 🤣  Tests pass for me locally, and I see "all checks have passed" on this PR.  Where is it failing?

**Comment by MrPetovan** [comment]
Oh, never mind, it's failing with the current code. 🤦‍♂️

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**
